### PR TITLE
Made Add Dataset Organisation an Autocomplete

### LINKED ISSUE #2328: Add Dataset "People" Section - Organisation Dropdown
As a user, when I add a dataset I want to be able to select the organisation that published the dataset.

This relates to the "What organisation is responsible for publishing this data" dropdown on the "People and Production" page of the "Add a dataset" flow.

image
from https://app.zeplin.io/project/5cff0c2b6308535da4f2dfe8/screen/5cff0da41f236d16148d163b

### Acceptance Criteria
- Should be able to be defaulted to a certain organisation - e.g. DAWR should have DAWR as the default value
- Typing in the box should cause a dropdown of known organisations from the search to appear
- You should still be able to put in whatever organisation you like if it isn't already in the system
- If a new organisation is entered, it should be created when the dataset is saved
- If an entered value matches an existing value exactly (ignoring case), add it as an organisation to the registry (and hence it'll be picked up by the search)

### Out of Scope For Now
- Selecting organisational unit
- What if the use types something very similar to an organisation that's already been added? Let it happen, we can revisit this if/when it becomes a problem.

### PR DESCRIPTION
**Pull request by AlexGilleran**
### What this PR does

Fixes #2328 

This turns the organisation field from a blank text input to a pre-populated autocomplete field.

This means:
- The generic parts of the vocabulary autocomplete have been pulled out into a base component used by both vocabulary autocomplete and the new organisation autocomplete
- I found some weird problems with the autocomplete not triggering in certain situations, so I've simplified the way that it debounces
- The new org autocomplete uses the same autocomplete API for organisations as the search facet dropdown
- When selected, we store the name of the organisation and the id. The user can also put in plain text. If there's no id when the dataset is created, the organisation is looked up in the search. If there's already an org with that name, it's matched, otherwise a new org is created
- It's also possible to set a default organisation id through helm (e.g. DAWR would set DAWR as their default org). This necessitated making building the initial state for the add dataset screens async (because we've got an id but we've got to look it up in the registry). I've created a HOC that displays "Loading..." until this is retrieved. This needs to be replaced with a proper loading icon.

Also:
- Hacked a fix to a problem with CKAN redirection where it always failed because there was no tenant id - now it's set to always be 0 (will work with single tenant deployments), needs to work for multi tenant

TODO:
  - [x] Create an issue for making the CKAN redirection work with multi-tenant properly (#2343)
  - [x] Make an issue for using the registry to check for Organisation duplication at the end of the add dataset process (#2337)
  - [x] Make an issue for fixing up the loading icons (#2344)

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)

### CODE REVIEW

**Comment by AlexGilleran** [comment]
Ok should be ready to test now - when you create a new dataset on the preview namespaces it should default to Department of Environment and Energy :)

File: magda-web-client/src/Components/Editing/AutoCompleteInput.tsx [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    render() {
+        const inputProps: AutosuggestInputProps<T> = {
+            ...this.props.inputProps,
+            onKeyUp: this.onKeyUp,
+            onChange: this.onChange,
+            onBlur: this.onBlur,
+            value: this.state.value
+        };
+
+        console.log(this.state.value);
```
Comment (t83714) [inline magda-web-client/src/Components/Editing/AutoCompleteInput.tsx]: do we need this?

File: magda-web-client/src/Components/Dataset/Add/DatasetAddMetadataPage.tsx [hunk 2]
```diff
@@ -724,6 +639,62 @@ class NewDataset extends React.Component<Prop, State> {
             datasetUsage,
             datasetAccess
         } = this.state;
+
+        if (!dataset.publisher) {
+            throw new Error("No publisher selected");
+        }
+
+        console.log(dataset.publisher);
```
[the remark this answers is not in the document]
Comment (AlexGilleran) [reply magda-web-client/src/Components/Dataset/Add/DatasetAddMetadataPage.tsx]: Oops 😬

File: magda-web-client/src/Components/Dataset/Add/Pages/People/DatasetAddPeoplePage.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                <h4>
+                    Which organisation is responsible for publishing this
+                    dataset?
+                </h4>
+                <div>
+                    <OrganisationAutoComplete
+                        defaultValue={dataset.publisher}
+                        onOrgSelected={editDataset("publisher")}
+                    />
+                    {/* <AlwaysEditor
```
[the remark this answers is not in the document]
Comment (AlexGilleran) [reply magda-web-client/src/Components/Dataset/Add/Pages/People/DatasetAddPeoplePage.tsx]: Oops 😬

File: magda-web-client/src/Components/Editing/AutoCompleteInput.tsx (hunk 1 above)
Comment (AlexGilleran) [reply magda-web-client/src/Components/Editing/AutoCompleteInput.tsx]: Oops 😬

**Comment by AlexGilleran** [comment]
Thanks @t83714 , removed those :)

**Pull request merged**